packages/sdk/src/lib/dataProtectorSharing/addToCollection.ts

```diff
@@ -83,17 +83,16 @@ export const addToCollection = async ({
     sharingContractAddress,
   });
 
-  const hasContractAlreadyBeenApproved = !approveTx;
-  // If no transaction, it means the dataProtectorSharing Contract has already been approved
-  if (hasContractAlreadyBeenApproved) {
-    vOnStatusUpdate({
-      title: 'APPROVE_COLLECTION_CONTRACT',
-      isDone: true,
-      payload: {
-        approveTxHash: approveTx?.hash,
-      },
-    });
-  }
+  vOnStatusUpdate({
+    title: 'APPROVE_COLLECTION_CONTRACT',
+    isDone: true,
+    payload: approveTx?.hash
+      ? { approveTxHash: approveTx.hash }
+      : {
+          message:
+            'Your ProtectedData has already been approved for the smart contract',
+        },
```

Comment:
I'm not sure we should write a message like this in English, the client will probably set its own message.
What we need is to give them something easy to check for, like a boolean.

```suggestion
      : { isAlreadyApproved: true },
```

or both:

```suggestion
      : {
          isAlreadyApproved: true,
          message:
            'Your ProtectedData has already been approved for the smart contract',
        },
```

I vote for this last option. The message adds more details, let's keep it.